packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FlutterWebView.m

```diff
@@ -348,6 +348,12 @@ - (NSString*)applySettings:(NSDictionary<NSString*, id>*)settings {
     } else if ([key isEqualToString:@"userAgent"]) {
       NSString* userAgent = settings[key];
       [self updateUserAgent:[userAgent isEqual:[NSNull null]] ? nil : userAgent];
+    } else if ([key isEqualToString:@"limitsNavigationsToAppBoundDomains"]) {
+      if (@available(iOS 14.0, *)) {
+        NSNumber* limitsNavigationsToAppBoundDomains = settings[key];
+        _webView.configuration.limitsNavigationsToAppBoundDomains =
+            [limitsNavigationsToAppBoundDomains boolValue];
+      }
```

Comment:
There's no test coverage of this part of the change; if this were accidentally removed in the future, the feature would stop working but no tests would fail. The best option here would be a native unit test that sends a message with this enabled, and ensures (probably via an exposed-for-testing accessor to give you the `_webView`) that the configuration is set correctly.

Reply:
I understand why this needs a unit test, but I'm not sure how to implement testing for it. I tried looking at the other properties, and to me, it looks like none of the properties would be caught in a test if native code were altered. So I'm not entirely sure how to proceed with the unit test.

Some of the questions I have regarding the native unit test. 

- Should it be in the flutter_webview_wkwebview package or the flutter_webview package?
- Should it be all swift code, or is there some way to test from flutter with flutter code sending the message over the message channel?
- Also if the test is added to the ios folder, is there anything special I need to do to make it work with the GitHub workflows/checks?

Reply:
> it looks like none of the properties would be caught in a test if native code were altered.

That's true, but: https://github.com/flutter/flutter/wiki/Plugin-Tests#faq-do-i-need-to-add-tests-even-if-the-part-of-the-code-im-changing-isnt-already-tested

> * Should it be in the flutter_webview_wkwebview package or the flutter_webview package?

The former. Unit tests should live in the same package as the code they are testing.

> * Should it be all swift code, or is there some way to test from flutter with flutter code sending the message over the message channel?

It should be Objective-C; testing from Dart would require an integration test, which would be much more involved for this case.

> * Also if the test is added to the ios folder, is there anything special I need to do to make it work with the GitHub workflows/checks?

Native unit tests are already configured for the package, you just need to add a test to those existing test files (or if you make a new file, add it to the same target).